### PR TITLE
Add a Vagrantfile to do local testing

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "centos/7"
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+
+  config.vm.provision "shell", inline: <<-SHELL
+    yum -y install epel-release
+    yum -y install puppet5-release || yum -y install https://yum.puppetlabs.com/puppet5/puppet5-release-el-7.noarch.rpm
```
Comment (evgeni) [inline Vagrantfile]: when would the first part succeed?

File: manifests/default.pp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+node 'jenkins-master' {
```
Comment (evgeni) [inline manifests/default.pp]: this file is only relevant for the vagrant tests, yet it creates a new prominent folder and is placed there. kinda feels it should be rather `vagrant/manifests/default.pp`?

File: Vagrantfile (hunk 1 above)
Comment (ekohl) [reply Vagrantfile]: On the second run. I often did a `vagrant rsync && vagrant provision` to apply changes.

Comment (evgeni) [reply Vagrantfile]: ah, fair

File: manifests/default.pp (hunk 2 above)
Comment (ekohl) [reply manifests/default.pp]: Updated. Also moved the Vagrantfile.

**Comment by ekohl** [comment]
I have moved back Vagrantfile to the root. The benefit is that the created VMs will be named `foreman-infra_jenkins-master` instead of `vagrant_jenkins-master` in libvirt which makes them easier to trace.

File: vagrant/manifests/default.pp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  class { 'web::base':
+    letsencrypt => false,
+  }
+
+  class { 'web::jenkins':
+    hostname => $facts['fqdn'],
+  }
+
+  package { 'python-pip':
```
Comment (evgeni) [inline vagrant/manifests/default.pp]: `ensure_packages(['python-pip'])`

Comment (ekohl) [reply vagrant/manifests/default.pp]: Did this in the JJB module since it's not only needed in the vagrant case.

**Comment by tbrisker** [comment]
I'm seeing this trying to run it locally:
```
Error: Evaluation Error: Unknown function: 'ensure_packages'. (file: /tmp/vagrant-puppet/modules-3354d8c37858ffbd1ef99488693bd0ab/jenkins_master/manifests/init.pp, line: 196, column: 3) on node jenkins-master

**Comment by evgeni** [comment]
> ```
> Error: Evaluation Error: Unknown function: 'ensure_packages'. (file: /tmp/vagrant-puppet/modules-3354d8c37858ffbd1ef99488693bd0ab/jenkins_master/manifests/init.pp, line: 196, column: 3) on node jenkins-master
> ```

`ensure_packages` comes from stdlib, which is in our submodules. Did you init/update the submodules in your checkout?

**Comment by tbrisker** [comment]
ignore me

> > ```
> > Error: Evaluation Error: Unknown function: 'ensure_packages'. (file: /tmp/vagrant-puppet/modules-3354d8c37858ffbd1ef99488693bd0ab/jenkins_master/manifests/init.pp, line: 196, column: 3) on node jenkins-master
> > ```
> 
> `ensure_packages` comes from stdlib, which is in our submodules. Did you init/update the submodules in your checkout?

me-- :man_facepalming:

**Pull request merged**